### ISSUE TITLE
Check why llama-cpp-python fails some tests for llama-cpp-python >= 0.2.38

### ISSUE THREAD

**Issue by dtiarks** [outlines-dev/outlines#638]
Some llama integration tests (e.g. `test_llamacpp_various_regexes`) fail for llama-cpp-python >= 0.2.38.

Investigate this further.

**Comment by lapp0** [comment]
Issue appears to be due to their speculative decoding change

https://github.com/abetlen/llama-cpp-python/commit/fb762a60411f53278454b4e9888c5bd9712d3779#diff-ed675dab7bf6bd468dd59d2b195d999021715f673252e51122d25edd8b9ade1eR537-R540

```
(Pdb) idx
8
(Pdb) self._input_ids
array([    1,  1824,   349,   272,  1141,   302,  5459,  9955, 28804],
      dtype=int32)
(Pdb) len(self._input_ids)
9
(Pdb) self._input_ids[:idx]
array([   1, 1824,  349,  272, 1141,  302, 5459, 9955], dtype=int32)
```

It cuts off the last token, `28804`, before applying the logits processor.

**Comment by lapp0** [comment]
PR here: https://github.com/abetlen/llama-cpp-python/pull/1179

**Comment by rlouf** [comment]
Solved with the latest release of `llama-cpp-python`. Thank you @lapp0!
